stdlib/public/runtime/Metadata.cpp

```diff
@@ -56,19 +62,45 @@
 using namespace swift;
 using namespace metadataimpl;
 
+// allocate memory up to a nearby page boundary
+static void *swift_allocPage(size_t size) {
```

Comment:
Could you rename this function pair to `swift_allocateMetadataRoundingToPage`/`swift_freeMetadata`?  Using "page" in the name makes it hard to interpret the parameter (is it size in bytes?  is it the number of pages?)  Also, we are passing the metadata tag on OS X, so this function should not be used for anything but metadata allocation.

I also see callers rounding the allocation size to a page size.  If this is critical for the function to operate correctly, could you move that rounding into the function?